core/src/main/java/com/amplifyframework/predictions/models/SyntaxToken.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.models;
+
+import androidx.annotation.NonNull;
+
+import java.util.Objects;
+
+/**
+ * Class that holds the syntax detection results for a
+ * string of text for the predictions category.
+ */
+public final class SyntaxToken {
```

Comment:
Javadoc makes it look like it should be named as one of the `Predictions*Result`  classes. But this is just a single token, in a collection of tokens, inside of a result, right?